core/web/apiv2/auth.py

```diff
@@ -82,30 +111,60 @@ async def __call__(self, user: User = Depends(get_current_user)) -> User:
 # API Endpoints
 router = APIRouter()
 
-
-@router.post("/token")
-async def login(response: Response, form_data: OAuth2PasswordRequestForm = Depends()):
-
-    if not YETI_AUTH:
-        user = UserSensitive.find(username="yeti")
-        if not user:
-            user = UserSensitive(username="yeti", admin=True)
-            user.set_password("yeti")
-            user.save()
-    else:
-        user = UserSensitive.find(username=form_data.username)
-        if not (user and user.verify_password(form_data.password)):
-            raise HTTPException(
-                status_code=status.HTTP_401_UNAUTHORIZED,
-                detail="Incorrect username or password",
-                headers={"WWW-Authenticate": "Bearer"},
-            )
-
-    access_token = create_access_token(
-        data={"sub": user.username}, expires_delta=ACCESS_TOKEN_EXPIRE_MINUTES
-    )
-    response.set_cookie(key="yeti_session", value=access_token, httponly=True)
-    return {"access_token": access_token, "token_type": "bearer"}
+# We only want certain endpoints to be defined depending on the auth module.
+if AUTH_MODULE == 'oidc':
+    @router.get('/oidc-login')
+    async def login_info(request: Request):
+        redirect_uri = request.url_for('oidc_callback')
+        redirect_uri = redirect_uri.replace(netloc='localhost:8000')
+        return await get_oauth_client().oidc.authorize_redirect(request, redirect_uri)
+
+
+    @router.get('/oidc-callback', response_class=RedirectResponse)
+    async def oidc_callback(request: Request) -> RedirectResponse:
+        try:
+            token = await get_oauth_client().oidc.authorize_access_token(request)
+        except OAuthError as e:
+            raise e
```

Comment:
Here you handle the exception :)

Reply:
Actually, I'm just re-raising the exception, so not really handling it :) I'll remove this bit of code.